### ISSUE TITLE
Threading timing issue

### ISSUE THREAD

**Issue by SmileyChris** [spesmilo/electrum#1054]
Installed electrum 2.0 on my laptop and fired it up, noticing the following exception being raised. Wouldn't synchronise and trying to bring up the network dialog caused the app to lock up.
```
Exception in thread Thread-13:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/chris/.virtualenvs/electrum/local/lib/python2.7/site-packages/electrum/network.py", line 362, in process_requests_thread
    self.process_request(request)
  File "/home/chris/.virtualenvs/electrum/local/lib/python2.7/site-packages/electrum/network.py", line 391, in process_request
    self.interface.send_request(request)
  File "/home/chris/.virtualenvs/electrum/local/lib/python2.7/site-packages/electrum/interface.py", line 290, in send_request
    self.pipe.send(r)
AttributeError: 'TcpInterface' object has no attribute 'pipe'
```

As a totally hacky test (and it turned out to work) in `network.py`, I just added a small sleep in to ensure the network threads had time to initialize. I'm sure there's a cleaner way so I'm not going to pull request this hack :smile_cat: 

```python
    def process_requests_thread(self):
        time.sleep(.1)
```

**Comment by ecdsa** [comment]
is there a way to trigger that bug reliably?

**Comment by SmileyChris** [comment]
Not that I can think of - timing issues are a pain to debug :-/

It triggers every time on my laptop, but on my desktop it never does.
Both installed on Ubuntu 14.10 in the same way.

**Comment by ecdsa** [comment]
yeah there should be a way to insert a sleep() that triggers it

**Comment by ecdsa** [comment]
can you try that please:
```
--- a/lib/network.py
+++ b/lib/network.py
@@ -355,6 +355,9 @@ class Network(threading.Thread):
 
     def process_requests_thread(self):
         while self.is_running():
+            if not self.is_connected():
+                time.sleep(0.1)
+                continue
             try:
                 request = self.requests_queue.get(timeout=0.1)
             except Queue.Empty:
```

**Comment by SmileyChris** [comment]
Ok, checked that out -- no error there but now I can't reproduce the error in the first place.

Without understanding the code at all, I'm wondering if it was only an issue if there's a request in the queue right off the bat. In which case, I *guess* your code would fix it... :)

**Comment by ecdsa** [comment]
well, unfortunately that patch breaks other things..

**Comment by SmileyChris** [comment]
Well the issue resolved itself for me, but probably only because I worked
around the issue. I'll let you have a ponder as to whether there's an easy
solution or if this is just an outlier bug. Thanks for electrum, you rock.

**Comment by ecdsa** [comment]
fixed in 73d8ba24d94cc1fb36c646635efecb39b89d0c9f
